tests/Backend/Workspaces/MetadataFromExasolWorkspaceTest.php

```diff
@@ -0,0 +1,332 @@
+            'KBC.datatype.nullable' => '',
+            'KBC.datatype.basetype' => 'DATE',
+            'KBC.datatype.length' => '10',
+        ];
+        // check that the new table has the correct metadata
+        $table = $this->_client->getTable($tableId);
+
+        $this->assertEquals([], $table['metadata']);
+        $this->assertArrayHasKey('string', $table['columnMetadata']);
+        $this->assertMetadata($expectedStringMetadata, $table['columnMetadata']['string']);
+        $this->assertArrayHasKey('char', $table['columnMetadata']);
+        $this->assertMetadata($expectedCharMetadata, $table['columnMetadata']['char']);
+        $this->assertArrayHasKey('integer', $table['columnMetadata']);
+        $this->assertMetadata($expectedIntegerMetadata, $table['columnMetadata']['integer']);
+        $this->assertArrayHasKey('decimal', $table['columnMetadata']);
+        $this->assertMetadata($expectedDecimalMetadata, $table['columnMetadata']['decimal']);
+        $this->assertArrayHasKey('real', $table['columnMetadata']);
+        $this->assertMetadata($expectedRealMetadata, $table['columnMetadata']['real']);
+        $this->assertArrayHasKey('double', $table['columnMetadata']);
+        $this->assertMetadata($expectedDoubleMetadata, $table['columnMetadata']['double']);
+        $this->assertArrayHasKey('boolean', $table['columnMetadata']);
+        $this->assertMetadata($expectedBooleanMetadata, $table['columnMetadata']['boolean']);
+        $this->assertArrayHasKey('time', $table['columnMetadata']);
+        $this->assertMetadata($expectedTimeMetadata, $table['columnMetadata']['time']);
+        $this->assertArrayHasKey('date', $table['columnMetadata']);
+        $this->assertMetadata($expectedDateMetadata, $table['columnMetadata']['date']);
+    }
+
+    public function testCopyImport()
+    {
+        $this->markTestSkipped('missing addTableColumn and incremental');
+        $table_id = $this->_client->createTable(
+            $this->getTestBucketId(self::STAGE_IN),
+            'languages3',
+            new CsvFile(__DIR__ . '/../../_data/languages.csv'),
+            array('primaryKey' => 'id')
+        );
+
+        // create workspace and source table in workspace
+        $workspace = $this->initTestWorkspace(self::BACKEND_EXASOL);
+
+        $tableId = 'Languages3';
+
+        $backend = WorkspaceBackendFactory::createWorkspaceBackend($workspace);
+        $backend->dropTableIfExists($tableId);
+
+        $connection = $workspace['connection'];
+        $db = $this->getDbConnection($connection);
+
+        $quotedTableId = $db->getDatabasePlatform()->quoteIdentifier(sprintf(
+            '%s.%s',
+            $connection['schema'],
+            $tableId
+        ));
+
+        $db->query("create table $quotedTableId (
+                \"id\" number(3,0) not null,
+                \"name\" varchar(50) default 'honza' not null 
+            );");
+        $db->query("insert into $quotedTableId ([id], [name]) values (1, 'cz');");
```

Comment:
hele to je taky nejaka synapsovina ne?

Reply:
Jo stejně jak https://github.com/keboola/storage-api-php-client/pull/654#discussion_r693694970